src/System.Private.ServiceModel/tests/Common/Infrastructure/src/ConditionalWcfTest.cs

```diff
@@ -0,0 +1,23 @@
+﻿using System;
+namespace Infrastructure.Common
+{
+    public class ConditionalWcfTest
+    {
+        public static bool IsTestEnabled(string testCategoryName)
+        {
+            string propertyValue = TestProperties.GetProperty(testCategoryName);
+            bool isEnabled;
+            if (!String.IsNullOrEmpty(propertyValue) && bool.TryParse(propertyValue, out isEnabled))
+            {
+                return isEnabled;
+            }
+
+            return false;
+        }
+
+        public static bool Domain_Joined_Tests_Enabled()
+        {
+            return IsTestEnabled(TestProperties.Domain_Joined_Tests_Enabled_PropertyName);
```

Comment:
I don't think so but it's a good question.  If an invalid name is passed in TestProperties will throw, and it should be obvious from the call stack which test had a problem.  Just to be safe though, I'll make that helper method private and leave only the well-formed ones public.  They won't even build if they try to use an invalid name.